frontend/src/utils/shares.ts

```diff
@@ -205,17 +207,94 @@ const getPurchaseResult = (
 
   return {
     shares: Shares.fromCollateralUnits(market.denom, purchasedShares),
-    price: buyAmountTotalCoins.dividedBy(purchasedShares),
-    liquidity: Coins.fromUnits(buyAmountTotalCoins.denom, liquidity),
-    fees: Coins.fromUnits(buyAmountTotalCoins.denom, fees),
+    price: coinsAmount.dividedBy(purchasedShares),
+    liquidity: Coins.fromUnits(market.denom, liquidity),
+    fees: Coins.fromUnits(market.denom, fees),
+  }
+}
+
+interface SaleResult {
+  coins: Coins
+  liquidity: Coins
+  fees: Coins
+  price: Coins
+}
+
+const getSaleResult = (
```

Comment:
@lvn-rusty-dragon I just defined this function and renamed a few variables, but I don't know how the math should work